$NetBSD: patch-aa,v 1.5 2014/07/14 15:30:10 fhajny Exp $

Add support for NetBSD.
Remove -fomit-frame-pointer on SunOS.

--- Makefile.orig	2014-07-12 14:40:52.000000000 +0000
+++ Makefile
@@ -89,7 +89,7 @@ DOCDIR = $(PREFIX)/doc/haproxy
 # Use TARGET=<target_name> to optimize for a specifc target OS among the
 # following list (use the default "generic" if uncertain) :
 #    generic, linux22, linux24, linux24e, linux26, solaris,
-#    freebsd, openbsd, cygwin, custom, aix51, aix52
+#    freebsd, netbsd, openbsd, cygwin, custom, aix51, aix52
 TARGET =
 
 #### TARGET CPU
@@ -107,7 +107,6 @@ ARCH =
 
 #### Toolchain options.
 # GCC is normally used both for compiling and linking.
-CC = gcc
 LD = $(CC)
 
 #### Debug flags (typically "-g").
@@ -267,7 +266,7 @@ ifeq ($(TARGET),solaris)
   # This is for Solaris 8
   # We also enable getaddrinfo() which works since solaris 8.
   USE_POLL       = implicit
-  TARGET_CFLAGS  = -fomit-frame-pointer -DFD_SETSIZE=65536 -D_REENTRANT
+  TARGET_CFLAGS  = -DFD_SETSIZE=65536 -D_REENTRANT
   TARGET_LDFLAGS = -lnsl -lsocket
   USE_TPROXY     = implicit
   USE_LIBCRYPT    = implicit
@@ -288,6 +287,12 @@ ifeq ($(TARGET),osx)
   USE_TPROXY     = implicit
   USE_LIBCRYPT   = implicit
 else
+ifeq ($(TARGET),netbsd)
+  # This is for NetBSD
+  USE_POLL       = implicit
+  USE_KQUEUE     = implicit
+  USE_TPROXY     = implicit
+else
 ifeq ($(TARGET),openbsd)
   # This is for OpenBSD >= 3.0
   USE_POLL       = implicit
@@ -318,6 +323,7 @@ endif # cygwin
 endif # aix52
 endif # aix51
 endif # openbsd
+endif # netbsd
 endif # osx
 endif # freebsd
 endif # solaris
@@ -634,7 +640,7 @@ all:
 	@echo "Please choose the target among the following supported list :"
 	@echo
 	@echo "   linux2628, linux26, linux24, linux24e, linux22, solaris"
-	@echo "   freebsd, openbsd, cygwin, custom, generic"
+	@echo "   freebsd, netbsd, openbsd, cygwin, custom, generic, aix52"
 	@echo
 	@echo "Use \"generic\" if you don't want any optimization, \"custom\" if you"
 	@echo "want to precisely tweak every option, or choose the target which"
